marshal.go

```diff
@@ -347,6 +378,27 @@ func (e *Encoder) marshal(v interface{}) error {
 	return fmt.Errorf("cannot marshal value of type %v", typ.Kind())
 }
 
+func (e *Encoder) encFloat(i float64) error {
+	if _, err := e.w.Write([]byte{FLOAT_SIGN}); err != nil {
+		return err
+	}
+	floatStr := fmt.Sprintf("%g", i)
+	size := len(floatStr) + 5
+	switch {
+	case math.IsNaN(i):
+		e.w.WriteString("\bnan")
+	case math.IsInf(i, 1):
+		e.w.WriteString("\binf")
+	case math.IsInf(i, -1):
+		e.w.WriteString("\t-inf")
+	default:
+		e.w.WriteString(fmt.Sprintf("%c", size))
```

Comment:
should we put lines 385-386 in here since we don't use floatStr or size except in default?